kratos/processes/find_intersected_geometrical_objects_with_obb_process.cpp

```diff
@@ -0,0 +1,455 @@
+
+    // We create new properties for debugging
+    if (mDebugOBB) {
+        this->GetModelPart1().CreateNewProperties(1001);
+        this->GetModelPart1().CreateSubModelPart(this->GetModelPart1().Name() + "_AUXILIAR_DEBUG_OBB");
+        this->GetModelPart2().CreateNewProperties(1002);
+        this->GetModelPart2().CreateSubModelPart(this->GetModelPart2().Name() + "_AUXILIAR_DEBUG_OBB");
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsWithOBBProcess::SetOctreeBoundingBox()
+{
+    // Getting first iterators
+    const auto it_node_begin_1 = BaseType::mrModelPartIntersected.NodesBegin();
+    const auto it_node_begin_2 = BaseType::mrModelPartIntersecting.NodesBegin();
+
+    // Setting initial guess
+    PointType low(it_node_begin_1->Coordinates());
+    PointType high(it_node_begin_1->Coordinates());
+
+    // Loop over all nodes in first modelpart
+    for (IndexType i_node = 0 ; i_node < BaseType::mrModelPartIntersected.NumberOfNodes(); ++i_node) {
+        auto it_node = it_node_begin_1 + i_node;
+        const array_1d<double,3>& r_coordinates = it_node->Coordinates();
+        for (IndexType i = 0; i < 3; i++) {
+            low[i] = r_coordinates[i] < low[i] ? r_coordinates[i] : low[i];
+            high[i] = r_coordinates[i] > high[i] ? r_coordinates[i] : high[i];
+        }
+    }
+
+    // Loop over all skin nodes
+    for (IndexType i_node = 0 ; i_node < BaseType::mrModelPartIntersecting.NumberOfNodes(); ++i_node) {
+        auto it_node = it_node_begin_2 + i_node;
+        const array_1d<double,3>& r_coordinates = it_node->Coordinates();
+        for (IndexType i = 0; i < 3; i++) {
+            low[i] = r_coordinates[i] < low[i] ? r_coordinates[i] : low[i];
+            high[i] = r_coordinates[i] > high[i] ? r_coordinates[i] : high[i];
+        }
+    }
+
+    // Slightly increase the bounding box size to avoid problems with geometries in the borders
+    // Note that std::numeric_limits<double>::double() is added for the 2D cases. Otherwise, the
+    // third component will be 0, breaking the octree behaviour.
+    for(IndexType i = 0 ; i < 3; i++) {
+        low[i] -= std::abs(high[i] - low[i])*1e-3 + std::numeric_limits<double>::epsilon();
+        high[i] += std::abs(high[i] - low[i])*1e-3 + std::numeric_limits<double>::epsilon();
+    }
+
+    // In case we consider the bounding box we extend box
+    if (mBoundingBoxFactor > 0.0) {
+        const std::size_t dimension = BaseType::mrModelPartIntersected.GetProcessInfo()[DOMAIN_SIZE];
+        for(IndexType i = 0 ; i < dimension; i++) {
+            if (std::abs(high[i] - low[i]) < mBoundingBoxFactor) {
+//                 low[i] -= mBoundingBoxFactor;
+                high[i] += mBoundingBoxFactor;
+            }
```

Comment:
I would suggest to inflate the bounding box by min and max and not only max point. This gives robustness issues specially when min point is origin

Reply:
OK

Reply:
For some reason this doesn't appear as updated/outdated